app/assets/javascripts/discourse/app/components/sidebar/user/custom-sections.js

```diff
@@ -23,31 +153,9 @@ export default class SidebarUserCustomSections extends Component {
   }
 
   get sections() {
-    this.currentUser.sidebarSections.forEach((section) => {
-      if (!section.public || this.currentUser.staff) {
-        section.headerActions = [
-          {
-            action: () => {
-              return showModal("sidebar-section-form", { model: section });
-            },
-            title: I18n.t("sidebar.sections.custom.edit"),
-          },
-        ];
-      }
-      section.decoratedTitle =
-        section.public && this.currentUser.staff
-          ? htmlSafe(`${iconHTML("globe")} ${section.title}`)
-          : section.title;
-      section.links.forEach((link) => {
-        if (!link.external) {
-          const routeInfoHelper = new RouteInfoHelper(this.router, link.value);
-          link.route = routeInfoHelper.route;
-          link.models = routeInfoHelper.models;
-          link.query = routeInfoHelper.query;
-        }
-      });
+    return this.currentUser.sidebarSections.map((section) => {
+      return new Section(section, this.currentUser, this.router);
```

Comment:
```suggestion
      return new Section({ section, currentUser: this.currentUser, router: this.router });
```

Minor suggestion but in the long run as we add more params, passing in an object makes the code much easier to read and we don't have to care about the order of the params.